File: examples/src/main/java/edu/tum/cs/isabelle/examples/Hello_PIDE.java
package edu.tum.cs.isabelle.examples.java;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import edu.tum.cs.isabelle.api.*;
import edu.tum.cs.isabelle.japi.*;
import edu.tum.cs.isabelle.setup.*;

public class Hello_PIDE {

  public static void main(String args[]) {
    Environment env = JSetup.makeEnvironment(JSetup.defaultSetup(new Version("2015")));
    JResources res = JResources.dumpIsabelleResources();
    Configuration config = res.makeConfiguration(Arrays.<Path> asList(), "Protocol");
    JSystem sys = JSystem.create(env, config);
    String response = sys.invoke(Operations.HELLO, "world");
    System.out.println(response);
    sys.dispose();
  }

}

File: modules/examples/src/main/scala/Hello_PIDE.scala
package info.hupel.isabelle.examples.scala

import scala.concurrent._
import scala.concurrent.duration._

import monix.execution.Scheduler.Implicits.global

import info.hupel.isabelle._
import info.hupel.isabelle.api._
import info.hupel.isabelle.setup._

object Hello_PIDE extends App {

  val setup = Setup.default(Version("2016")).right.get // yolo
  val resources = Resources.dumpIsabelleResources().right.get // yolo
  val config = resources.makeConfiguration(Nil, Nil, "Protocol")

  val transaction =
    for {
      env <- setup.makeEnvironment(config)
      sys <- System.create(env, config)
      response <- sys.invoke(Operation.Hello)("world")
      _ = println(response.unsafeGet)
      () <- sys.dispose
    } yield ()

  Await.result(transaction, Duration.Inf)

}
